influxql/parser.go

```diff
@@ -687,10 +687,18 @@ func (p *Parser) parseSelectStatement(tr targetRequirement) (*SelectStatement, e
 		}
 	})
 
+	// If we have a group by interval, but no aggregate function, it's an invalid statement
 	if d, _ := stmt.GroupByInterval(); stmt.IsRawQuery && d > 0 {
 		return nil, fmt.Errorf("GROUP BY requires at least one aggregate function")
 	}
 
+	// If we have a count function with a group by time without a where clause, it's an invalid statement
+	if tr == targetNotRequired { // ignore create continuous query statements
+		if d, _ := stmt.GroupByInterval(); !stmt.IsRawQuery && d > 0 && !stmt.hasTimeDimensions(stmt.Condition) {
```

Comment:
You sure this can't return an error? `GroupByInternal()` returns 0 if there is no `GROUP BY`? Do also really need `IsRawQuery`? I am curious.